subsys/bluetooth/host/hci_core.c

```diff
@@ -7185,6 +7377,351 @@ bool bt_addr_le_is_bonded(uint8_t id, const bt_addr_le_t *addr)
 	}
 }
 
+int bt_le_per_adv_set_param(struct bt_le_ext_adv *adv,
+			    const struct bt_le_per_adv_param *param)
+{
+	struct bt_hci_cp_le_set_per_adv_param *cp;
+	struct net_buf *buf;
+	int err;
+
+	if (atomic_test_bit(adv->flags, BT_ADV_SCANNABLE)) {
+		return -EINVAL;
+	} else if (atomic_test_bit(adv->flags, BT_ADV_CONNECTABLE)) {
+		return -EINVAL;
+	} else if (!atomic_test_bit(adv->flags, BT_ADV_EXT_ADV)) {
+		return -EINVAL;
+	}
+
+	if (param->interval_min < 0x0006 ||
+	    param->interval_max < 0x0006 ||
```

Comment:
should interval_max be checked for upper bound instead?

Reply:
At least it doesn't make sense to check if max < 6, since we already check min < 6 and if max < min. I added the a max check, but that cannot possible fail as long as the variable type remains uint16 (as the maximum is in fact 0xFFFF). I assume that the compiler will remove the check when optimizing the code, so it's just there for the looks (and possible a safeguard if someone tries to change the type).